src/lib/validateFrontmatter.ts

```diff
@@ -16,7 +16,7 @@ export async function validateFrontmatter(
   pages: PageMetadata[],
   promptQuestion: string,
   outputDir?: string,
-): Promise<PageMetadata<GuidesRequestRepresentation>[]> {
+): Promise<PageMetadata<PageRequestSchema<typeof this.route>>[]> {
```

Comment:
Had no idea you could do `this` typings outside the scope of the function like this.